src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts

```diff
@@ -336,47 +389,112 @@ Parameters:
     t.expectOK(res);
   });
 
-g.test('depth_2d_coords')
+g.test('sampled_3d_coords,derivatives')
   .specURL('https://www.w3.org/TR/WGSL/#texturesample')
   .desc(
     `
-fn textureSample(t: texture_depth_2d, s: sampler, coords: vec2<f32>) -> f32
-fn textureSample(t: texture_depth_2d, s: sampler, coords: vec2<f32>, offset: vec2<i32>) -> f32
+fn textureSample(t: texture_3d<f32>, s: sampler, coords: vec3<f32>) -> vec4<f32>
+fn textureSample(t: texture_3d<f32>, s: sampler, coords: vec3<f32>, offset: vec3<i32>) -> vec4<f32>
+fn textureSample(t: texture_cube<f32>, s: sampler, coords: vec3<f32>) -> vec4<f32>
 
-Parameters:
- * t  The sampled, depth, or external texture to sample.
- * s  The sampler type.
- * coords The texture coordinates used for sampling.
- * offset
-    * The optional texel offset applied to the unnormalized texture coordinate before sampling the texture.
-    * This offset is applied before applying any texture wrapping modes.
-    * The offset expression must be a creation-time expression (e.g. vec2<i32>(1, 2)).
-    * Each offset component must be at least -8 and at most 7.
-      Values outside of this range will result in a shader-creation error.
+test mip level selection based on derivatives
+
+* TODO: test 3d compressed textures formats. Just remove the filter below 'viewDimension'
 `
   )
-  .paramsSubcasesOnly(u =>
+  .params(u =>
     u
-      .combine('S', ['clamp-to-edge', 'repeat', 'mirror-repeat'] as const)
-      .combine('coords', generateCoordBoundaries(2))
-      .combine('offset', generateOffsets(2))
+      .combine('format', kTestableColorFormats)
+      .filter(t => {
+        const type = kTextureFormatInfo[t.format].color?.type;
+        const canPotentialFilter = type === 'float' || type === 'unfilterable-float';
+        // We can't easily put random bytes into compressed textures if they are float formats
+        // since we want the range to be +/- 1000 and not +/- infinity or NaN.
+        const isFillable = !isCompressedTextureFormat(t.format) || !t.format.endsWith('float');
+        return canPotentialFilter && isFillable;
+      })
+      .combine('viewDimension', kCoord3DViewsForDerivativeTests)
```

Comment:
Side note: One thing I notice during gardening, maybe it's better to put `viewDimension` before `format` so that suppression can be more inclusive. E.g. cube map fails on a platform and it will be `viewDiemsnions='cube';*`
When format are put first the roller is likely to add suppressions for flaky tests scattered around the expectations.txt

However I don't know if changing this would make roller suffer for the next cts roll...